core/src/main/scala/org/apache/spark/broadcast/TorrentBroadcast.scala

```diff
@@ -90,8 +94,6 @@ private[spark] class TorrentBroadcast[T: ClassTag](obj: T, id: Long)
   /** Total number of blocks this broadcast variable contains. */
   private val numBlocks: Int = writeBlocks(obj)
 
-  /** Whether to generate checksum for blocks or not. */
-  private var checksumEnabled: Boolean = false
   /** The checksum for all the blocks. */
   private var checksums: Array[Int] = _
```

Comment:
The same applies to `checksums` also ?

Reply:
It doesn't look like checksums is set in setConf

Reply:
TL/DR; you are right, no change required for `checksums`.

The difference seems to be a nuance of `_` vs explicit value (`checksums` is initialized in `val nunBlocks = writeBlocks(obj)`, while declared after).

```
scala > class Test { 
     |   var a = test()
     |   var b: String = "123"
     |   var c: String = _
     |   def test(): Int = {
     |     b = "abc"
     |     c = "abc"
     |     1
     |   }
     | }
     
scala> val test = new Test()
val test: Test = Test@2184c27b

scala> test.a
val res11: Int = 1

scala> test.b
val res12: String = 123

scala> test.c
val res13: String = abc
```

Reply:
I think it's just ordering here. If setConf happens before the initializer of foo, and then foo is declared and initialized, the initializer overwrites whatever setConf did